storage/country.cpp

```diff
@@ -18,119 +18,147 @@ namespace storage
+  void InsertOldMwmMapping(TCountryId const & newId, TCountryId const & oldId) override
+  {
+    m_idsMapping[oldId].insert(newId);
+  }
 
-    if (addedNode != nullptr)
-      addedNode->SetSubtreeAttrs(mwmChildCounter, mwmChildSize);
+  void InsertAffiliation(TCountryId const & countryId, string const & affilation) override
+  {
+    m_affiliations.insert(make_pair(countryId, affilation));
   }
-  return make_pair(mwmCounter, mwmSize);
-}
 
-template <class ToDo>
-void LoadGroupTwoComponentMwmsImpl(int depth, json_t * group, TCountryId const & parent, ToDo & toDo)
+  TMappingOldMwm GetMapping() const override { return m_idsMapping; }
+};
+
+class StoreFile2InfoSingleMwms : public StoreSingleMwmInterface
 {
-  // @TODO(bykoianko) After we stop supporting two component mwms (with routing files)
-  // remove code below.
-  size_t const groupListSize = json_array_size(group);
-  for (size_t i = 0; i < groupListSize; ++i)
-  {
-    json_t * j = json_array_get(group, i);
+  TMappingOldMwm m_idsMapping;
+  map<string, CountryInfo> & m_file2info;
 
-    char const * file = json_string_value(json_object_get(j, "f"));
-    // if file is empty, it's the same as the name
-    if (!file)
-    {
-      file = json_string_value(json_object_get(j, "n"));
-      if (!file)
-        MYTHROW(my::Json::Exception, ("Country name is missing"));
-    }
+public:
+  StoreFile2InfoSingleMwms(map<string, CountryInfo> & file2info) : m_file2info(file2info) {}
 
-    // We expect that mwm and routing files should be less than 2GB.
-    uint32_t const mwmSize = static_cast<uint32_t>(json_integer_value(json_object_get(j, "s")));
-    uint32_t const routingSize = static_cast<uint32_t>(json_integer_value(json_object_get(j, "rs")));
-    toDo(file, mwmSize, routingSize, depth, parent);
+  // StoreSingleMwmInterface overrides:
+  Country * InsertToCountryTree(TCountryId const & id, uint32_t /* mapSize */, int /* depth */,
+                                TCountryId const & /* parent */) override
+  {
+    CountryInfo info(id);
+    m_file2info[id] = move(info);
+    return nullptr;
+  }
 
-    json_t * children = json_object_get(j, "g");
-    if (children)
-      LoadGroupTwoComponentMwmsImpl(depth + 1, children, file, toDo);
+  void InsertOldMwmMapping(TCountryId const & /* newId */, TCountryId const & /* oldId */) override
+  {
   }
-}
+  void InsertAffiliation(TCountryId const & /* countryId */,
+                         string const & /* affilation */) override
+  {
+  }
+  TMappingOldMwm GetMapping() const override
+  {
+    ASSERT(false, ());
+    return map<TCountryId, TCountriesSet>();
+  }
+};
+}  // namespace
 
-template <class ToDo>
-bool LoadCountriesSingleMwmsImpl(string const & jsonBuffer,TCountryId const & parent, ToDo & toDo)
+TMwmSubtreeAttrs LoadGroupSingleMwmsImpl(int depth, json_t * node, TCountryId const & parent,
+                                         StoreSingleMwmInterface & store)
 {
-  try
+  uint32_t mwmCounter = 0;
+  size_t mwmSize = 0;
```

Comment:
Please, move mwmCounter and mwmSize to the code where they're used. Now they're declared at the beginning of the functions but used 20LOCs later.